R/mcmc.R

```diff
@@ -60,87 +55,62 @@ setMethod(
   def = function(data,
                  model,
                  options,
-                 verbose = FALSE,
                  from_prior = data@nObs == 0L,
                  ...) {
-    assert_flag(verbose)
     assert_flag(from_prior)
 
-    if (verbose) {
-      futile.logger::flog.threshold(futile.logger::INFO, name = "mcmc")
-    } else {
-      futile.logger::flog.threshold(futile.logger::FATAL, name = "mcmc")
-    }
-
     jags_model_fun <- if (from_prior) {
       model@priormodel
     } else {
       h_join_models(model@datamodel, model@priormodel)
     }
+    jags_file <- h_write_model(jags_model_fun)
 
-    # Write the model into the file.
-    jags_dir <- file.path(tempdir(), "R_crmPack")
-    jags_file <- tempfile("jags_model_fun", jags_dir, ".txt")
-    # Don't warn, as the temp dir often exists (which is OK).
-    dir.create(jags_dir, showWarnings = FALSE)
-    futile.logger::flog.info("Temporary directory: %s", jags_dir, name = "mcmc")
-    h_write_model(jags_model_fun, jags_file)
-
-    # Get the initial values for the parameters.
     inits <- do.call(model@init, h_slots(data, formalArgs(model@init)))
     assert_list(inits)
     inits <- inits[sapply(inits, length) > 0L]
```

Comment:
maybe would be nice:
```suggestion
    inits <- h_get_model_inits(model, data)
```

Reply:
Yes, I agree!